boot/boot.go

```diff
@@ -163,16 +163,33 @@ func applicable(s snap.PlaceInfo, t snap.Type, dev Device) bool {
 	return true
 }
 
-// bootState exposes the boot state for a type of boot snap.
-type bootState interface {
+// earlyBootState exposes the boot state for a type of boot snap during the
+// early boot, i.e. during the initramfs or, strictly for testing purposes, the
+// bootloader stages.
+type earlyBootState interface {
+	// chooseSnapInitramfsMount chooses which snap should be mounted during the
+	// early boot sequence, i.e. the initramfs. actually committing the choice
+	// is done via the returned bootStateUpdate's commit method. Note that there
+	// may or may not actually be anything to commit, for example the kernel
+	// snap does not have anything to commit during the initramfs
+	chooseSnapInitramfsMount(bootStateUpdate) (sn snap.PlaceInfo, choices bootStateUpdate, err error)
```

Comment:
I'm not entirely sure this worth the complexity given that only one type has something to commit,  it's also strange it is its own interface but then it ends up implemented by the same structs just with a flag set on them